File: Question 0090 - Subsets II/src/Solution.java
import java.util.*;

class Solution {
    private void solve(List<List<Integer>> ans, List<Integer> subset, int[] nums, int start)
    {
        ans.add(new ArrayList<>(subset));

        for (int i = start; i < nums.length; i++) {
            if(i > start && nums[i] == nums[i-1]) continue;
            subset.add(nums[i]);
            solve(ans, subset, nums, i+1);
            subset.remove(subset.size()-1);
        }
    }
    public List<List<Integer>> subsetsWithDup(int[] nums)
    {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> subset = new ArrayList<>();
        Arrays.sort(nums);
        solve(ans,subset, nums, 0);
        return ans;
    }
}

File: Question 0090 - Subsets II/src/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
private:
    void solve(vector<int> &nums, vector<vector<int>> &ans, vector<int> &curr, int idx)
    {
        ans.push_back(curr);

        for (int i = idx; i < nums.size(); i++)
        {
            if(i > idx && nums[i] == nums[i-1]) continue;
            curr.push_back(nums[i]);
            solve(nums, ans, curr, i+1);
            curr.pop_back();
        }
    }

public:
    vector<vector<int>> subsetsWithDup(vector<int> &nums)
    {
        vector<vector<int>> ans;
        vector<int> curr;
        sort(nums.begin(), nums.end());
        solve(nums, ans, curr, 0);
        return ans;
    }
};